src/Release/GetPublicRelease.php

```diff
@@ -34,7 +35,10 @@ public function execute()
         $publicReleases = [ ];
 
         foreach ( $releases as $release ) {
-            if ( $release->isIsVisible() ) {
+            if ( !$release->isVisible() && $release->getReleaseDate()->getTimestamp() < time() ) {
```

Comment:
* Refactor this if condition into a method

* time() is global, so this should be inverted. \DateTimeInterface can be passed into the constructor